runtime/src/main/kotlin/kotlin/native/internal/KTypeParameterImpl.kt

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the LICENSE file.
+ */
+
+package kotlin.native.internal
+
+import kotlin.reflect.*
+
+internal data class KTypeParameterImpl(
+        override val name: String,
+        val containerFqName: String,
```

Comment:
Is it used?

Reply:
It is used in the `equals` and `hashCode`.

Reply:
I see, you are right.
But then I have another question: should `equals` and `hashCode` use `upperBounds`? This seems quite inefficient. Also not sure about `variance` or `isReified` either.
What does JVM implementation do?

Reply:
Good point. I copied this from JS implementation, but JVM only uses `name` and `container` for `equals`/`hashCode`, which is logical.